### ISSUE TITLE
FNX-5254 ⁃ [UX] Private Browsing

### ISSUE THREAD

**Issue by lime124** [mozilla-mobile/fenix#181]
GH issue to track design work. Nicole is the designer

**Fenix Flows WIP**
https://docs.google.com/drawings/d/11Kj7OAvit6bE6QQt_ven0xnHGhmwu4Jc_N_oazdryAo/edit

**Comment by bbinto** [comment]
- Design private browsing to be visually distinct
- TP on by default

**Comment by bbinto** [comment]
@topotropic - this would have to be done before milestone 2 for engineering start (Feb 8). 

Please comment add mock-ups as they are ready. Please also connect with @colintheshots on feasibility.

**Comment by topotropic** [comment]
Meta https://github.com/mozilla-mobile/fenix/issues/216

**Comment by bbinto** [comment]
- What is the user flow after hitting the private mode, this is blocking eng to proper estimate.
- How do you get out of private mode?
- Moving back and forth in between modes

Please let us know when this will be finalized?

**Comment by lime124** [comment]
Just FYI - I edited Barbara's comment just now to @ nicole and add in the notification comment from that eng meeting.

**Comment by pocmo** [comment]
Those screens looks great! Tbh I wish Fenix would always look like in those private mode screens. :)

I wonder how using light/dark themes to distinguish between normal/private browsing affects our battery meta (#240) and generic dark mode (#252)? If battery live is something we really want to have an impact on then it seems counter-productive to have a bright white app by default? (CC @colintheshots)

**Comment by brampitoyo** [comment]
@pocmo 

> I wonder how using light/dark themes to distinguish between normal/private browsing affects our battery meta and generic dark mode?

With the next version of Android rumored to support dark mode, it seems smart to **decouple light/dark themes with normal/private browsing modes**.

What we may want to consider instead is to have theme selection under Settings (which can be either “Light”, “Dark” or “Follow system settings”).

To mark private browsing, we colour it differently from other icons – neither light grey or dark blue, but purple.

Finding something that looks distinct from every other icons, but equally at home in both dark and light themes, is hard. Take the mockup below with a grain of salt.

<img width="360" src="https://user-images.githubusercontent.com/51007/52026548-1ece2480-256d-11e9-99a7-67b8cc3d1985.png">

Maybe something to consider in future versions?

**Comment by brampitoyo** [comment]
One question I’d like to pose to the team is whether we’d like Private Browsing history and tabs to persist or be erased/closed when the app is quit.

**Scenario**

* Open Fenix
* Enter Private Browsing
* Open 4 tabs
* Force quit Fenix on app switcher

**Question**
Should Private Browsing reopen its last state, or should it erase itself and start from scratch?

**On iOS Safari, Firefox and Chrome**

* Private Browsing history is not erased
* When reopened, the app remembers that you were last on Private Browsing mode
* The app opens the Private Tab that was last viewed before the app was force quit

**On Android Chrome and Firefox**

* Private Browsing history is erased
* When reopened, the app forgot that you were last on Private Browsing mode
* The app opens the normal tab that was last viewed before the app was forced to quit

**Proposed Fenix behaviour**

* Private browsing history is erased (like Android)
* When reopened, the app remembers that you were last on Private Browsing mode (like iOS)
* The app opens an empty Private Browsing Home Screen

In summary, when Fenix is force quit while in Private Browsing, it will erase your Private Browsing history, but will remember that you still want Private Mode to open first.

**Comment by sblatz** [comment]
>When there are multiple tabs, you need to scroll down a bit to access the “Delete Session” button. It’s a bit cumbersome with all the scrolling, but the UI is consistent. We’ll design a more accessible way to delete session in subsequent versions.

@brampitoyo is there a reason we can't just have the "delete session" button just pinned to the bottom of the screen and the tabs just appear from beneath it as you scroll up? That way the button is always visible and you can still see lots of tabs at once.

**Comment by brampitoyo** [comment]
@sblatz wrote:

> @brampitoyo is there a reason we can't just have the "delete session" button just pinned to the bottom of the screen and the tabs just appear from beneath it as you scroll up?

Yes, we can – and thanks for pointing it out!

In fact, the first set of concepts that I’ve explored has an “Erase” FAB on the tab list view (you can ignore the rest of the UI):

<img width="360" src="https://user-images.githubusercontent.com/51007/52182065-0e17fa00-285e-11e9-8cda-790ac7797a1f.png">

We can return to this UI again after v1 has released.

**Comment by sblatz** [comment]
@brampitoyo one more question for you: do you have a specific transition animation in mind for switching the home view from normal to private mode (namely the background gradient)

**Comment by bbinto** [comment]
@brampitoyo love the design and flow!

> One question I’d like to pose to the team is whether we’d like Private Browsing history and tabs to persist or be erased/closed when the app is quit.

I echo your suggestion for Fenix: "In summary, when Fenix is force quit while in Private Browsing, it will erase your Private Browsing history, but will remember that you still want Private Mode to open first."

**Comment by sblatz** [comment]
@brampitoyo how do we want the browser menu to look in private mode? This is how it is by default:

<img width=300 src=https://user-images.githubusercontent.com/4400286/52493943-8bef4480-2b81-11e9-9eea-da59b72c35a3.png>
